### ISSUE TITLE
Alternative Linux packaging solutions

### ISSUE THREAD

**Issue by intergrav** [modrinth/code#497]
At the time of writing, there are only two official ways for using the Modrinth App on Linux - AppImages and Debian packages. While this is fine for people on Debian-based distributions, and those who like using AppImages, some may prefer to use a package manager such as `flatpak` or `dnf` as it's easier to manage packages that way. According to [this comment](https://github.com/modrinth/theseus/issues/497#issuecomment-1666737825), the current compiled Debian packages aren't very great right now because they don't include proper dependency info and leave out some important stuff.

Along with AppImages and Debian packages, there are various packaging solutions to consider. If there's other packaging types that you think would be a good idea, suggest them in this thread. I personally think that the Flatpak should be maintained by Modrinth, and the rest could be maintained by members of the community if they wanted to do so. I think this is especially important as the Modrinth App is now in an [open beta](https://blog.modrinth.com/p/launcher).

This issue serves as a place to discuss packaging, and I'll also treat this as a sort of "unofficial download page" for Linux users with updated information on what packages to use and where to get them.

### ✅
  1. a package exists and works properly
  2. that package is considered ready for daily use
  3. that package is available on commonly used repositories, such as a Flatpak being on Flathub for example

### 🔃
  1. same as 1 and 2 in the above
  2. however, that package is not available on commonly used repositories, but is available on a developer's personal repository. this may not be applicable for some solutions and will use the checkmark icon instead, such as COPR

### ✅ <img alt="Flathub logo" src="https://raw.githubusercontent.com/flatpak/flatpak/main/flatpak.png" height="18px"> [Flatpak](https://flatpak.org/)

Package is available and maintained on [Flathub](https://flathub.org/apps/com.modrinth.ModrinthApp) - `com.modrinth.ModrinthApp` - thank you @getchoo and other contributors!

<details>

Package that can be used on most common Linux distributions, and also good for security as Flatpaks are sandboxed unlike other types. This will be the package that most Linux users are looking for.

###### Notable mentions/issues

- https://github.com/tauri-apps/tauri/issues/3619 (@Hyphrio)
- May be tricky to upload to Flathub, see [this comment](https://github.com/modrinth/theseus/issues/497#issuecomment-1666737825)

###### Who's working on this?

- @getchoo is currently working on a package https://github.com/modrinth/theseus/pull/559
- @getchoo opened a PR for Flathub https://github.com/flathub/flathub/pull/4950

</details>

### ✅ <img src="https://wiki.installgentoo.com/images/f/f9/Arch-linux-logo.png" height="18px"> [Arch](https://archlinux.org/)

Packages are available and maintained on the [AUR](https://aur.archlinux.org/packages/modrinth-app) - `modrinth-app`, `modrinth-app-bin`, and `modrinth-app-appimage` - thank you [Antti](https://aur.archlinux.org/account/Antti)!

<details>

###### Any ideas on this?

- @offbeat-stuff suggested making a PKGBUILD that extracts the `deb` package to be used as an Arch pacman package. [Here's an example](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=linux-xanmod-linux-bin-x64v3)

###### Notable mentions/issues

- https://github.com/tauri-apps/tauri/issues/3728 (@Hyphrio)
- https://github.com/tauri-apps/tauri/pull/4301 (@Hyphrio)

###### Who's working on this?

- [Antti](https://aur.archlinux.org/account/Antti) has made two AUR packages
  - `modrinth-app`
  - `modrinth-app-bin`
- [Nixuge](https://aur.archlinux.org/account/Nixuge) has made two AUR packages, however they could be broken as they miss many dependencies
  - `modrinth-app-git`
  - `modrinth-cli-git`

</details>

### ✅ <img alt="Nix logo" src="https://upload.wikimedia.org/wikipedia/commons/2/28/Nix_snowflake.svg" height="16px"> [Nix](https://nixos.org/)

Package is available and maintained on [Nixpkgs](https://search.nixos.org/packages?show=modrinth-app) - `modrinth-app` - thank you @getchoo and other contributors!

<details>

Nix can be used on both Linux and Darwin systems and is the package manager for NixOS.

###### Who's working on this?

- @getchoo has a PR in nixpkgs https://github.com/NixOS/nixpkgs/pull/289149
- @getchoo is also working on a Nix flake https://github.com/modrinth/theseus/pull/561

</details>

### ✅ <img alt="Homebrew logo" src="https://upload.wikimedia.org/wikipedia/commons/9/95/Homebrew_logo.svg" width="22px" height="22px"> [Homebrew](https://brew.sh)

Package is available and maintained on [Homebrew Cask](https://formulae.brew.sh/cask/modrinth) - `modrinth`.

<details>

Homebrew is a package manager for MacOS (and Linux). Idea from https://github.com/modrinth/theseus/issues/560. It would allow for easy installation on MacOS.

</details>

### ✅ <img alt="Gentoo logo" src="https://upload.wikimedia.org/wikipedia/commons/4/48/Gentoo_Linux_logo_matte.svg" height="18px"> [Gentoo](https://wiki.gentoo.org/wiki/Project:GURU)

Package is available and maintained on [Gentoo GURU](https://wiki.gentoo.org/wiki/Project:GURU) - `modrinth-app-bin` - thank you @Norbiros and other contributors!

<details>

User-maintained package repository for Gentoo.

</details>

### <img alt="Fedora logo" src="https://cdn.freebiesupply.com/logos/large/2x/fedora-1-logo-png-transparent.png" height="18px"> [Fedora Copr](https://copr.fedorainfracloud.org/)

<details>

User-maintained package repositories for Fedora, CentOS Stream, and RHEL. Can be enabled with `dnf copr enable <author>/<repo>`

###### Notable mentions/issues

- https://github.com/tauri-apps/tauri/issues/4402 (@Hyphrio)
- May be difficult, see [this comment](https://github.com/modrinth/theseus/issues/497#issuecomment-1666737825)

</details>

### 🔃 🍨 [Scoop](https://scoop.sh)

Package is available and maintained on [brawaru/bucket](https://github.com/brawaru/bucket) - thank you @brawaru!

<details>

Scoop is a command-line installer for Windows. It has many benefits, such as installing packages in a portable way (by default, it installs packages to `~/scoop`). Package manifests are just simple JSON files - this shouldn't be too hard to set up.

</details>

### ✅ <img alt="WinGet logo" src="https://raw.githubusercontent.com/microsoft/winget-cli/master/.github/images/WindowsPackageManager_Assets/ICO/PNG/_48.png" height="18px"> [WinGet](https://learn.microsoft.com/en-us/windows/package-manager/)

Package is available on [WinGet](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Modrinth/ModrinthApp) - currently [looking for maintainers](https://github.com/modrinth/code/issues/497#issuecomment-2679735144)!

<details>

WinGet is Microsoft's own package manager.

</details>

## Other information

Theseus is very complex to package and doing so for many different formats at a time would be a little chaotic. I think that the common package types like Flatpak, Arch and Nix packages should be done.

Leave your thoughts below!

###### For discussion, see the [thread](https://discord.com/channels/734077874708938864/1138510665947041876) in the `#development` channel of the [Modrinth Discord](https://discord.gg/modrinth). You can also continue discussing here if you prefer that.

**Comment by Hyphrio** [comment]
For Linux these two Tauri issues are related:
https://github.com/tauri-apps/tauri/issues/4402
https://github.com/tauri-apps/tauri/issues/3619

**Comment by offbeat-stuff** [comment]
At least make a PKGBUILD that extracts the deb package to be used as an archlinux pacman package.

**Comment by offbeat-stuff** [comment]
Here is a PKGBUILD for archlinux that extracts a .deb package

https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=linux-xanmod-linux-bin-x64v3

**Comment by getchoo** [comment]
> At least make a PKGBUILD that extracts the deb package to be used as an archlinux pacman package.

this isn't a good method for the long term, especially as arch is known to make abi incompatible changes to packages frequently, or at the very least push versions of libraries usually incompatible with those found on debian/ubuntu based distros (there are also problems between those, but usually less so). [here's a good example](https://github.com/PrismLauncher/PrismLauncher/issues/991)

for a temporary thing it would probable be fine? but ideally all community packages should be built against their distro's packages to ensure the final binaries actually work

**Comment by getchoo** [comment]
> I personally think that the Flatpak should be maintained by Modrinth, and the rest could be maintained by members of the community if they wanted to do so.

i 100% agree with this. it's very hard and completely impractical for upstream to support the endless amounts of linux distributions out there; universal solutions should be the main focus here, along with basic files for distribution that can be used there and in downstream packages

> [Flatpak](https://flatpak.org/)/[Flathub](https://flathub.org/)

the biggest issue here is that networking may need to be enabled during building due to flatpak's build tools lacking support for pnpm lock files. this is a problem since flathub doesn't really like that and it could make it very difficult for it to be uploaded there

> [Nixpkgs](https://search.nixos.org/packages)

this is pretty much complete and fully working on any linux distro. macos support for is [being worked on](https://github.com/getchoo/nix-exprs/pull/9). the problem with pnpm and offline builds also showed up here, but since nix is a lot more flexible with it's build process, i was able to figure out a way to maintain reproducibility while still using pnpm's lock file. i will try to upstream this in nixpkgs once a stable release is made

> [Fedora Copr](https://copr.fedorainfracloud.org/)
> [Arch User Repository](https://aur.archlinux.org/)
> [Alpine Linux packages](https://pkgs.alpinelinux.org/packages)

like you said, these should probably be community packages. i would like to note though that some may be difficult (mainly fedora rpms) due to the issue with pnpm and offline builds as mentioned before. arch is very lose with this though so it should work fine; i can't say anything about alpine since i've never used/packaged for it personally

the only other thing i would like to note is that tauri's native package export feature *really* shouldn't be used here. the current debian package for example seems to strip out the other tools theseus provides such as it's cli, doesn't include dependency information (i'm assuming the direct dependencies of theseus are compiled statically...which isn't great but works ig) which can cause issues with even launching minecraft, and in general is very inflexible for the future where manual intervention for packaging may be required. i'm shocked the deb package works in it's current state, let alone if it can even actually start minecraft ootb. we should be following the standard of other linux packages (specifically packages for other launchers) by supporting the formats themselves instead of relying on a tool that in all honestly, doesn't produce good packages. most of this will fall on the community of individual distros, but for a download link on the upstream maintainer's website, poorly auto generated debs and appimages aren't really acceptable for "linux support" imo. the best solution to fix this is flatpak

**Comment by Hyphrio** [comment]
Tauri PR https://github.com/tauri-apps/tauri/pull/4301 seems to be related for Pacman but seems to have stalled

**Comment by CompeyDev** [comment]
I'd be willing to draft `PKGBUILD`s for the AUR, just lmk.

**Comment by TheoCGaming** [comment]
If you do go for Flatpak, please, PLEASE include proper permissions. I'm tired of having to deal with broken permissions on Flatpak packages.

**Comment by MagmaBro123** [comment]
> 

Flatseal can help with fixing those, but agreed.

**Comment by intergrav** [comment]
> Flatseal can help with fixing those, but agreed.

The user shouldn't have to use Flatseal to fix broken permissions provided by the package, that's just bad design

**Comment by MagmaBro123** [comment]
> > Flatseal can help with fixing those, but agreed.
> 
> The user shouldn't have to use Flatseal to fix broken permissions provided by the package, that's just bad design

Again, I said I agreed

**Comment by blt-r** [comment]
There is a [tool](https://github.com/flatpak/flatpak-builder-tools/tree/master/node) to generate flatpak builder sources definitions from `package-lock.json` or `yarn.lock`, but it doesn't support `pnpm-lock.yaml`.  The only way I found to convert `pnpm-lock.yaml` to something that the tool can consume is [this npm package](https://www.npmjs.com/package/pnpm-lock-export), but I didn't manage to make it work. We need to get either yarn or npm lockfile with right dependency versions. (Or implement support for pnpm in that tool, but that's very not easy).

Besides that, we probably will need to disable auto updating for the flatpak package. If there is no better way, that could be done with some `sed` in the build script.

Since the whole app is bundled in a single executable, packaging it as a flatpak would probably be pretty easy.

**Comment by blt-r** [comment]
Actually, on a closer look, the resulting binary has quite a few dynamic dependencies. I'll try to see if I can put all of them into the flatpak

**Comment by intergrav** [comment]
I just made a forum thread in the Modrinth Discord server for discussing about this: https://discord.com/channels/734077874708938864/1138510665947041876
I'll also edit the issue and place it around the top, you can continue discussing here aswell :)

**Comment by AnOpenSauceDev** [comment]
3 days and still no mention of `snap`... 
/s

**Comment by MagmaBro123** [comment]
Ew, Snaps.

**Comment by getchoo** [comment]
> https://discord.com/channels/734077874708938864/1138510665947041876

quoting a message i made here:

nix packaging for linux and darwin is basically complete, just needs a few little touchups. i have flatpak almost working, but as it stands it would only be able to be distributed through flathub - only the website. this is due to no flatpak build tools supporting pnpm’s lockfile format; it’s a similar case with nix, but there are a few diy methods already in nixpkgs i was able to use so not an issue there. for flatpak, networking needs to be enabled during the build but that’s a big nono

only other “issue” is that tauri (or what i’m hoping to be only the specific version of tauri being used) is using an old version of webkitgtk to build so…i need to compile all of webkitgtk for it to work, which causes my pc to oom

it seems that upstream is willing to switch to npm ([source](https://discord.com/channels/734077874708938864/1138510665947041876/1138547341780074576)) which solves the first issues; everything else with the flatpak seems pretty good.

one thing i thought of though was making some packaging guidelines for the app, as currently some things such as actual naming can be unclear. maybe this could be done with upstream sources for stuff like desktop files?

**Comment by MagmaBro123** [comment]
> > https://discord.com/channels/734077874708938864/1138510665947041876
> 
> quoting a message i made here:
> 
> nix packaging for linux and darwin is basically complete, just needs a few little touchups. i have flatpak almost working, but as it stands it would only be able to be distributed through flathub - only the website. this is due to no flatpak build tools supporting pnpm’s lockfile format; it’s a similar case with nix, but there are a few diy methods already in nixpkgs i was able to use so not an issue there. for flatpak, networking needs to be enabled during the build but that’s a big nono
> 
> only other “issue” is that tauri (or what i’m hoping to be only the specific version of tauri being used) is using an old version of webkitgtk to build so…i need to compile all of webkitgtk for it to work, which causes my pc to oom
> 
> it seems that upstream is willing to switch to npm ([source](https://discord.com/channels/734077874708938864/1138510665947041876/1138547341780074576)) which solves the first issues; everything else with the flatpak seems pretty good.
> 
> one thing i thought of though was making some packaging guidelines for the app, as currently some things such as actual naming can be unclear. maybe this could be done with upstream sources for stuff like desktop files?

You and the Modrinth team would probably be better at chatting about how this would work

**Comment by AnOpenSauceDev** [comment]
> And, Snaps also suck, Debs are kinda the stone-age of package formats, and AppImages are kinda annoying due to you having to manually download each new AppImage per update. Also, both the Deb and AppImage Modrinth provides seem to be quite buggy. At least, afaik.

I was joking about snaps (unfortunately Canonical isn't), and I've had tonnes of issues with _installing_ the DEB version at all, plus packaging systems that bundle everything like AppImages and Flatpak tend to take up more space than native (10x more for the modrinth app), which might be an issue for some people.

**Comment by MagmaBro123** [comment]
> > And, Snaps also suck, Debs are kinda the stone-age of package formats, and AppImages are kinda annoying due to you having to manually download each new AppImage per update. Also, both the Deb and AppImage Modrinth provides seem to be quite buggy. At least, afaik.
> 
> I was joking about snaps (unfortunately Canonical isn't), and I've had tonnes of issues with _installing_ the DEB version at all, plus packaging systems that bundle everything like AppImages and Flatpak tend to take up more space than native (10x more for the modrinth app), which might be an issue for some people.

Eh, true. But there's no real universal way of packaging apps easily.

**Comment by getchoo** [comment]
> Flatpak tend to take up more space than native (10x more for the modrinth app),

this is basically a made up number. in fact, most of modrinth app's dependencies are *already in the runtime* - this means unlike some flatpaks, there are *very* few dependencies specific to it that need to be bundled. it should be a similar size to the deb + the size of the runtime

**Comment by MagmaBro123** [comment]
https://www.youtube.com/watch?v=IYXlgzrZRIE

**Comment by AnOpenSauceDev** [comment]
> > Flatpak tend to take up more space than native (10x more for the modrinth app),
> 
> this is basically a made up number. in fact, most of modrinth app's dependencies are _already in the runtime_ - this means unlike some flatpaks, there are _very_ few dependencies specific to it that need to be bundled. it should be a similar size to the deb + the size of the runtime

I was more talking about the appimage, I'm aware that flatpak has measures to keep it's size down.

**Comment by cryolithic** [comment]
I'll add that flatpak is what the Steam Deck is using, so that's a big gaming base there

**Comment by AnOpenSauceDev** [comment]
> I'll add that flatpak is what the Steam Deck is using, so that's a big gaming base there

flatpak is supported by almost all modern distros in general, but the deck can techically run the AppImage

**Comment by intergrav** [comment]
Added getchoo's two PRs to the issue

- https://github.com/modrinth/theseus/pull/559
- https://github.com/modrinth/theseus/pull/561

I've also added a section for this Homebrew issue:

- https://github.com/modrinth/theseus/issues/560

**Comment by offbeat-stuff** [comment]
no arm64 support sad

**Comment by offbeat-stuff** [comment]
ok fair other launchers dont support arm too

**Comment by AnOpenSauceDev** [comment]
> ok fair other launchers dont support arm too

MultiMC5 has a few ARM forks iirc.

> no arm64 support sad

it wouldn't be _impossible_ to compile this for an ARM64 machine, but most of these devices have a small userbase, have different natives that aren't mature, etc, etc. But I'm pretty sure you can just paste the `.mrpack` into MultiMC, and run it that way. <br>

you might be able to use the GUI (no clue about in-game performance) if you emulate via [UTM](https://github.com/utmapp/UTM) or [QEMU](https://github.com/qemu/qemu) on an M1 Mac, but clearly I'm not a mac user.

**Comment by getchoo** [comment]
> no arm64 support sad

the nix package can build for arm and has already in ci (see [here](https://garnix.io/build/Q9P8nJj9)). i don't have an arm desktop or anything to actually test it on, but it should work unless tauri is incompatible

**Comment by getchoo** [comment]
also posting this here from discord for aur builds:

i've made a patch for [`modrinth-app-git`](https://aur.archlinux.org/packages/modrinth-app-git) that should let it build. i don't run arch personally, so i can't really test it (hey twice i've said that now!), but anyone who runs arch might want to check it out and send it to the maintainer

<details>
  <summary>click for patch</summary>
  
  ```diff
diff --git a/PKGBUILD b/PKGBUILD
index c955fb2..44c7be8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,7 +2,7 @@
 
 pkgname=modrinth-app-git
 _gitname=theseus
-pkgver=r147.3f46882
+pkgver=v0.4.0.r3.g5f0d44a
 pkgrel=1
 pkgdesc='Official Modrinth launcher. Open-source, built by the community, for the community.'
 url=https://github.com/modrinth/theseus
@@ -10,34 +10,64 @@ arch=(x86_64)
 license=(GPL3)
 conflicts=('modrinth-app')
 provides=('modrinth-app')
-makedepends=('cargo-tauri' 'yarn' 'pnpm')
+makedepends=('cargo' 'pnpm' 'git')
+# tauri deps
+depends=('openssl' 'dbus' 'freetype2' 'gtk3' 'libappindicator-gtk3' 'librsvg' 'libsoup' 'webkit2gtk'
+	# minecraft deps
+	'libgl' 'libpulse' 'libx11' 'libxcursor' 'libxext' 'libxxf86vm')
+optdepends=(
+	'xorg-xrandr: for older minecraft versions'
+)
 source=("git+https://github.com/modrinth/theseus")
 sha256sums=('SKIP')
 
 
 pkgver() {
-	cd "${srcdir}/${_gitname}"
-	printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short=7 HEAD)"
+	cd "$_gitname"
+
+	git describe --long --tags --abbrev=7 | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+	cd "$_gitname"
+
+	export RUSTUP_TOOLCHAIN=stable
+	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+
+	cd "theseus_gui"
+	pnpm install --frozen-lockfile --no-optional
 }
 
 build() {
-	cd "${srcdir}/${_gitname}/theseus_gui/"
+	cd "$_gitname/theseus_gui/"
+
+	pnpm build
 
-	# Bundle the deb only as it takes no time (unlike the appimage)
-	# & contains every folder at their right place ready to go
-	sed -i -E \
-		"s|\"targets\": \"all\"|\"targets\": \"deb\"|" \
-		"src-tauri/tauri.conf.json"
+	cd ..
 
-	yarn install
-	cargo tauri build
+	export RUSTUP_TOOLCHAIN=stable
+	export CARGO_TARGET_DIR=target
+	cargo build --frozen --release --all-features
+}
+
+check() {
+	cd "$_gitname"
+
+	export RUSTUP_TOOLCHAIN=stable
+	cargo test --frozen --all-features
 }
 
 package() {
-	# Grab the folder name (has the version embeeded in it)
-	folder=$(find . -type d -name "modrinth-app*") 2>/dev/null
+	cd "$_gitname"
+
+	mkdir -p "$pkgdir"/usr/{bin,share/{applications,icons/hicolor/256x256/apps}}
+
+	pushd target/release
+	for bin in theseus_*; do
+		install -Dm755 "$bin" "$pkgdir"/usr/bin/"$bin"
+	done
+	popd
 
-	# Then just copy everything from the deb bundle folder
-	cp -r "${folder}/data/usr/" \
-		"${pkgdir}"
+	install -Dm644 theseus_gui/src-tauri/icons/Square284x284Logo.png "$pkgdir"/usr/share/icons/hicolor/256x256/apps/com.modrinth.ModrinthApp.png
+	#install -Dm644 distribution/com.modrinth.ModrinthApp.desktop "$pkgdir"/usr/share/applications/com.modrinth.ModrinthApp.desktop
 }
```
</summary>

**Comment by CompeyDev** [comment]
> 



> also posting this here from discord for aur builds:
> 
> i've made a patch for [`modrinth-app-git`](https://aur.archlinux.org/packages/modrinth-app-git) that should let it build. i don't run arch personally, so i can't really test it (hey twice i've said that now!), but anyone who runs arch might want to check it out and send it to the maintainer
> click for patch
> 
> ```diff
> diff --git a/PKGBUILD b/PKGBUILD
> index c955fb2..44c7be8 100644
> --- a/PKGBUILD
> +++ b/PKGBUILD
> @@ -2,7 +2,7 @@
> 
> pkgname=modrinth-app-git
> _gitname=theseus
> -pkgver=r147.3f46882
> +pkgver=v0.4.0.r3.g5f0d44a
> pkgrel=1
> pkgdesc='Official Modrinth launcher. Open-source, built by the community, for the community.'
> url=https://github.com/modrinth/theseus
> @@ -10,34 +10,64 @@ arch=(x86_64)
> license=(GPL3)
> conflicts=('modrinth-app')
> provides=('modrinth-app')
> -makedepends=('cargo-tauri' 'yarn' 'pnpm')
> +makedepends=('cargo' 'pnpm' 'git')
> +# tauri deps
> +depends=('openssl' 'dbus' 'freetype2' 'gtk3' 'libappindicator-gtk3' 'librsvg' 'libsoup' 'webkit2gtk'
> +	# minecraft deps
> +	'libgl' 'libpulse' 'libx11' 'libxcursor' 'libxext' 'libxxf86vm')
> +optdepends=(
> +	'xorg-xrandr: for older minecraft versions'
> +)
> source=("git+https://github.com/modrinth/theseus")
> sha256sums=('SKIP')
> 
> 
> pkgver() {
> -	cd "${srcdir}/${_gitname}"
> -	printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short=7 HEAD)"
> +	cd "$_gitname"
> +
> +	git describe --long --tags --abbrev=7 | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
> +}
> +
> +prepare() {
> +	cd "$_gitname"
> +
> +	export RUSTUP_TOOLCHAIN=stable
> +	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
> +
> +	cd "theseus_gui"
> +	pnpm install --frozen-lockfile --no-optional
> }
> 
> build() {
> -	cd "${srcdir}/${_gitname}/theseus_gui/"
> +	cd "$_gitname/theseus_gui/"
> +
> +	pnpm build
> 
> -	# Bundle the deb only as it takes no time (unlike the appimage)
> -	# & contains every folder at their right place ready to go
> -	sed -i -E \
> -		"s|\"targets\": \"all\"|\"targets\": \"deb\"|" \
> -		"src-tauri/tauri.conf.json"
> +	cd ..
> 
> -	yarn install
> -	cargo tauri build
> +	export RUSTUP_TOOLCHAIN=stable
> +	export CARGO_TARGET_DIR=target
> +	cargo build --frozen --release --all-features
> +}
> +
> +check() {
> +	cd "$_gitname"
> +
> +	export RUSTUP_TOOLCHAIN=stable
> +	cargo test --frozen --all-features
> }
> 
> package() {
> -	# Grab the folder name (has the version embeeded in it)
> -	folder=$(find . -type d -name "modrinth-app*") 2>/dev/null
> +	cd "$_gitname"
> +
> +	mkdir -p "$pkgdir"/usr/{bin,share/{applications,icons/hicolor/256x256/apps}}
> +
> +	pushd target/release
> +	for bin in theseus_*; do
> +		install -Dm755 "$bin" "$pkgdir"/usr/bin/"$bin"
> +	done
> +	popd
> 
> -	# Then just copy everything from the deb bundle folder
> -	cp -r "${folder}/data/usr/" \
> -		"${pkgdir}"
> +	install -Dm644 theseus_gui/src-tauri/icons/Square284x284Logo.png "$pkgdir"/usr/share/icons/hicolor/256x256/apps/com.modrinth.ModrinthApp.png
> +	#install -Dm644 distribution/com.modrinth.ModrinthApp.desktop "$pkgdir"/usr/share/applications/com.modrinth.ModrinthApp.desktop
> }
> ```

Checking it our right now.

**Comment by CompeyDev** [comment]
Apparently it's corrupt?

![image](https://github.com/modrinth/theseus/assets/74418041/f61f089d-fff9-4950-a6bc-4dbd686691dd)

**Comment by CompeyDev** [comment]
> Apparently it's corrupt?
> 
> ![image](https://user-images.githubusercontent.com/74418041/260057866-f61f089d-fff9-4950-a6bc-4dbd686691dd.png)

Oh I realized that there's a insertions/deletions summary on line 7 for some reason haha.

**Comment by CompeyDev** [comment]
It works perfectly fine!
![image](https://github.com/modrinth/theseus/assets/74418041/a5f3ac9d-0c82-4452-aba2-c1752081bc2d)

**Comment by AnOpenSauceDev** [comment]
> > no arm64 support sad
> 
> the nix package can build for arm and has already in ci (see [here](https://garnix.io/build/Q9P8nJj9)). i don't have an arm desktop or anything to actually test it on, but it should work unless tauri is incompatible

Currently testing this out on my Raspberry PI.

**Comment by AnOpenSauceDev** [comment]
after compiling for... _7 hours..._ `gio` failed to build and panicked. I'll try doing this again.

**Comment by CompeyDev** [comment]
> after compiling for... _7 hours..._ `gio` failed to build and panicked. I'll try doing this again.

I feel your pain. Truly.

**Comment by AnOpenSauceDev** [comment]
> I feel your pain. Truly.

This is the way of `rustc`...


> after compiling for... _7 hours..._ `gio` failed to build and panicked. **I'll try doing this again.** (huge mistake)

even with a ludicrous amount of swap space it failed yet again,  this time with a `glib` error.

<details>
<summary>Warning! very big ANSI text file!</summary>
```

^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m x11 v2.21.0
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m dunce v1.0.4
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m zstd-safe v5.0.2+zstd.1.5.2
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m glob v0.3.1
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m httpdate v1.0.2
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m tower-service v0.3.2
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m tauri-utils v1.4.0
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m tokio-util v0.7.8
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m pango v0.15.10
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m cairo-rs v0.15.12
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m h2 v0.3.20
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m atk v0.15.1
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m tokio-native-tls v0.3.1
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m gdkx11-sys v0.15.1
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m x11-dl v2.21.0
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m raw-window-handle v0.5.2
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m fdeflate v0.3.0
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m sha1 v0.10.5
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m hyper v0.14.27
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m hyper-tls v0.5.0
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m serde_urlencoded v0.7.1
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m tracing-log v0.1.3
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m sharded-slab v0.1.4
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m tao v0.16.2
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m encoding_rs v0.8.32
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m hex v0.4.3
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m mime v0.3.17
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m ipnet v2.8.0
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m base64 v0.21.2
^[[0m^[[0m^[[1m^[[32m   Compiling^[[0m reqwest v0.11.18
^[[0m^[[0m^[[1m^[[31merror^[[0m^[[1m:^[[0m could not compile `glib` (lib)

Caused by:
  process didn't exit successfully: `rustc --crate-name glib --edition=2021 /build/cargo-vendor-dir/glib-0.15.12/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,fut>
^[[0m^[[0m^[[1m^[[33mwarning^[[0m^[[1m:^[[0m build failed, waiting for other jobs to finish...```

**Comment by AnOpenSauceDev** [comment]
looks like --edition is being forced as 2021 when glib only has a 2018 or 2015 edition:
error: argument for `--edition` must be one of: 2015|2018. (instead was `2021`)

**Comment by getchoo** [comment]
> after compiling for... _7 hours..._ `gio` failed to build and panicked. I'll try doing this again.

build with `nix build --accept-flake-config github:getchoo/nix-exprs#modrinth-app`. this will let you use my binary cache to avoid compiling, since everything is reproducible (outside of hardware issues ofc :p)

**Comment by AnOpenSauceDev** [comment]
> build with `nix build --accept-flake-config github:getchoo/nix-exprs#modrinth-app`. this will let you use my binary cache to avoid compiling, since everything is reproducible (outside of hardware issues ofc :p)

When i try to run this, i get `error: unable to execute '/nix/store/08kixl87lyjkakvg98vfkvz0yf26dwig-modrinth-app-unstable-2023-08-06/bin/modrinth-app-unstable': No such file or directory`, and i think it might just be easier on my sanity to cross-compile or simulate a more powerful ARM device on my personal server.

**Comment by AnOpenSauceDev** [comment]
> ### <img alt="Nix logo" height="16px" src="https://camo.githubusercontent.com/aced8526cb6832145de2dab7645521bf168412dd298fe4491c7868fc570c967f/68747470733a2f2f75706c6f61642e77696b696d656469612e6f72672f77696b6970656469612f636f6d6d6f6e732f322f32382f4e69785f736e6f77666c616b652e737667"> [Nixpkgs](https://search.nixos.org/packages)
> 
> Large package repository. Nix can be used on both Linux and Darwin systems, and is the package manager for NixOS. While Flatpaks _can_ be used on NixOS, Flatpak goes against it's reproducibility.
> ###### Who's working on this?
> 
>     * @getchoo already has a working package. Will be in nixpkgs when Theseus is stable.
>       
>       * NixOS: `nix shell github:getchoo/nix-exprs#theseus` and then `theseus_gui`
>       * Non-NixOS and Darwin (untested, but should work): `nix shell github:getchoo/nix-exprs#theseus` and then `nix run --impure github:guibou/nixGL -- theseus_gui`
> 
>     * @getchoo is also working on a Nix flake [feat: add nix flake #561](https://github.com/modrinth/theseus/pull/561)

@intergrav `theseus` is now `modrinth-app`. [Source](https://github.com/NixOS/nixpkgs/issues/247468#issuecomment-1669964328), you might want to change `nix shell github:getchoo/nix-exprs#theseus`  to `nix shell github:getchoo/nix-exprs#modrinth-app` to reflect on that.

**Comment by getchoo** [comment]
> > build with `nix build --accept-flake-config github:getchoo/nix-exprs#modrinth-app`. this will let you use my binary cache to avoid compiling, since everything is reproducible (outside of hardware issues ofc :p)
> 
> ~ When i try to run this, i get `error: unable to execute '/nix/store/08kixl87lyjkakvg98vfkvz0yf26dwig-modrinth-app-unstable-2023-08-06/bin/modrinth-app-unstable': No such file or directory`, and i think it might just be easier on my sanity to cross-compile or simulate a more powerful ARM device on my personal server. ~

i think this might be because of the naming of the package and nix thinking it's also the name of the binary. try doing `nix shell --accept-flake-config github:getchoo/nix-exprs#modrinth-app` and then run `modrinth-app`

**Comment by AnOpenSauceDev** [comment]
> i think this might be because of the naming of the package and nix thinking it's also the name of the binary. try doing `nix shell --accept-flake-config github:getchoo/nix-exprs#modrinth-app` and then run `modrinth-app`

I figured that out  not too long after, but the natives are _still building_ from the `nix run` after _9 hours_

**Comment by intergrav** [comment]
> @intergrav `theseus` is now `modrinth-app`. [Source](https://github.com/NixOS/nixpkgs/issues/247468#issuecomment-1669964328), you might want to change `nix shell github:getchoo/nix-exprs#theseus` to `nix shell github:getchoo/nix-exprs#modrinth-app` to reflect on that.

Should I also change `theseus-gui` to `modrinth-app-gui`?

**Comment by TheoCGaming** [comment]
Probably a good idea so it's a little easier to search for.

**Comment by getchoo** [comment]
> Should I also change theseus-gui to modrinth-app-gui?

no, the main binary is installed as `modrinth-app`

nixos users should also be able to just select the desktop icon now if it's installed through `nix profile`, home manager, a system configuration, etc.

**Comment by AnOpenSauceDev** [comment]
> Should I also change `theseus-gui` to `modrinth-app-gui`?

No, the `gui` package is still called `theseus`, at least for now.

**Comment by triphora** [comment]
When packaged, the GUI should always be modrinth-app. If we decide to launch the CLI at a later date, that will be modrinth-app-cli.

**Comment by AnOpenSauceDev** [comment]
I managed to compile and run `modrinth-app`. the app doesn't crash at the very least on ARM, and it seems to be working well. The only caveat is that I was having some issues with nixGL, so I'm just going to install NixOS and try again.

**Comment by TheoCGaming** [comment]
AUR package "modrinth-app-git" is missing rustup and the PKGBUILD needs a line to ensure rustup is properly configured

proposed changes to PKGBUILD:
[PKGBUILD-modrinthapp.txt](https://github.com/modrinth/theseus/files/12345046/PKGBUILD-modrinthapp.txt)
[PKGBUILD-modrinthapp.diff.txt](https://github.com/modrinth/theseus/files/12345050/PKGBUILD-modrinthapp.diff.txt)
github's file upload is dumb

**Comment by TheoCGaming** [comment]
waitaminute
the diff is wrong

**Comment by TheoCGaming** [comment]
[PKGBUILD-modrinthapp.diff.txt](https://github.com/modrinth/theseus/files/12345087/PKGBUILD-modrinthapp.diff.txt)
fixed

**Comment by MagmaBro123** [comment]
> 
I would love to see a flatpak version, it would make a lot of things more simple and for the end user it would be easier to understand to install. Persoanlly I want to as i can more easily set its launch env as running in hybrid mode getting it to use the correct gpu is annoying

I'd recommend checking out the thread for this stuff on the Discord.

**Comment by Geometrically** [comment]
FWIW we are fine to switch to NPM for development if it makes linux package easier. if PNPM is a limitation we are fine dropping it

**Comment by getchoo** [comment]
pnpm has mostly been worked around (or permanently solved in the case of nix) so i wouldn't consider it needed just yet. the bigger issue here would probably be with tauri's dependency on an older version of webkitgtk that has caused us to fall back to the gnome 42 runtime, since my own machine *and* gha are running into OOMs from compiling webkitgtk; the ci time before it fails has also reached close to 5 hours. there isn't really anything we can do here as even the latest stable version of tauri is still dependent on this, so the biggest help would probably be upgrading to tauri 2.0.0 as soon as it's stable

**Comment by Boxersteavee** [comment]
Oh and don't forget the deb package requires a deprecated library that is no longer in ubuntu's apt repository for 22.04. I had to go to their website and find the 20.04 version, which luckily still worked.

**Comment by intergrav** [comment]
Update: I'm now including Windows packaging in this issue, although I've noted it as low priority

**Comment by osfanbuff63** [comment]
For a Scoop package, we'd pretty much need a portable version of Theseus, which as far as I know isn't possible easily with Tauri but I could be wrong.

**Comment by intergrav** [comment]
> For a Scoop package, we'd pretty much need a portable version of Theseus

I don't think the package *has* to be portable, although I only really install CLI programs with Scoop which are portable, and other software with WinGet, so I wouldn't know. I assume this because they have a `nonportable` bucket with a few apps that I know for sure don't have portable versions

**Comment by AnOpenSauceDev** [comment]
I don't do much with Windows anymore, but Chocolatey exists.

**Comment by getchoo** [comment]
> I don't think the package has to be portable

this is ture. the prism launcher package for example was using the regular, non portable version of the package for a while

**Comment by osfanbuff63** [comment]
Alright, I'll look into that.

**Comment by intergrav** [comment]
I've added my own Scoop package to the issue

**Comment by CompeyDev** [comment]
> I've added my own Scoop package to the issue

Bravo!

**Comment by jptrzy** [comment]
I did create a Alpine Linux package that works.

Thou minecraft won't run because of alpine not using java with gnu core utils.
It looks like, if you want to play minecraft on that distro you **need to use flatpak**,
which I learned only after compiling theseus too many times 😢

[Modrinth-App APKBUILD](https://github.com/modrinth/theseus/files/13329609/modrinth-app.zip)
[modrinth-app-0.6.1-r0.apk](https://github.com/modrinth/theseus/files/13329606/modrinth-app-0.6.1-r0.zip)

**Comment by Danik1601** [comment]
> It looks like, if you want to play minecraft on that distro you **need to use flatpak**, which I learned only after compiling theseus too many times 😅

They have been [actively working](https://github.com/modrinth/theseus/pull/559) on it, but the work has stalled being blocked by Tauri dependencies. It seems that the work on Flatpak version will continue after Tauri 2.0 is released (after Q1 2024)

**Comment by DerpyPlayz18** [comment]
The issue for [tauri apps as RPM](https://github.com/tauri-apps/tauri/issues/4402) has been recently closed and compatibility has been added. Now modrinth can be packaged for Fedora.

**Comment by getchoo** [comment]
i've opened a PR for nixpkgs; flathub is next
https://github.com/NixOS/nixpkgs/pull/289149

**Comment by RokeJulianLockhart** [comment]
> https://github.com/modrinth/theseus/issues/497#issue-1838041992

@intergrav, don't merely package RPMs for Fedora - make them compatible with openSUSE (and others) too, using universal RPM packaging, as https://forums.opensuse.org/t/proton-bridge-installation-error/172223/5 explains. I'd also build them at the OBS (as https://github.com/chidiwilliams/buzz/issues/466#issuecomment-1773311903 explains).

**Comment by getchoo** [comment]
> The issue for https://github.com/tauri-apps/tauri/issues/4402 has been recently closed and compatibility has been added. Now modrinth can be packaged for Fedora.

this shouldn't be used. it will have the same issues as the current deb package, as tauri has no idea it's meant to also launch minecraft

> make them compatible with openSUSE (and others) too, using universal RPM packaging

this can get messy very fast. i've attempted it before with prismlauncher, and while it did work, it made the spec files more much difficult to maintain with little benefit. i would highly advise against anyone doing this, especially if you are also targeting leap releases as well as tumbleweed or fedora + enterprise linux.

> I'd also build them at the OBS (as https://github.com/chidiwilliams/buzz/issues/466#issuecomment-1773311903 explains).

this is a good idea though!...just not for fedora. [copr](https://copr.fedorainfracloud.org/coprs/) is much better for fedora users with it's built in support, and likewise obs is great for suse users with it's built in support but not so great on fedora. in my experience, users of both distros also overwhelmingly prefer their "native" platforms

**Comment by getchoo** [comment]
> i've opened a PR for nixpkgs; flathub is next [NixOS/nixpkgs#289149](https://github.com/NixOS/nixpkgs/pull/289149)

for anyone who's missed it: the flathub pr is up! you can find the link in the original post for this thread

**Comment by MagmaBro123** [comment]
> > i've opened a PR for nixpkgs; flathub is next [NixOS/nixpkgs#289149](https://github.com/NixOS/nixpkgs/pull/289149)
> 
> for anyone who's missed it: the flathub pr is up! you can find the link in the original post for this thread

Still has a bunch of unfixed stuff though

**Comment by getchoo** [comment]
> Still has a bunch of unfixed stuff though

it works fine outside of bugs found in webkitgtk (so nothing specific to the package). anything mentioned there is only about meeting flathub's guidelines rather than getting a working flatpak period

**Comment by MagmaBro123** [comment]
aH.

**Comment by intergrav** [comment]
> I did create a Alpine Linux package that works.
> 
> Thou minecraft won't run because of alpine not using java with gnu core utils. It looks like, if you want to play minecraft on that distro you **need to use flatpak**, which I learned only after compiling theseus too many times 😅
> 
> PS. I don't see any reason using alpine package at the current state of alpine java package, but I will include my efforts bellow 😁
> 
> [Modrinth-App APKBUILD](https://github.com/modrinth/theseus/files/13329609/modrinth-app.zip) [modrinth-app-0.6.1-r0.apk](https://github.com/modrinth/theseus/files/13329606/modrinth-app-0.6.1-r0.zip)

@jptrzy you think I should just remove the mention of an Alpine package in this issue considering that it doesn't make sense to use it in it's current state?

I'll remove it for now but lmk if you want me to add it back to the issue.

**Comment by intergrav** [comment]
I'm currently working on cleaning up and updating the issue with latest info

**Comment by Enderteck** [comment]
Hi why isn't the Flathub package on the Modrinth website ?

It still downloads the appimage by default and there isn't any kind of flatpak support outside of [third party packages](https://support.modrinth.com/en/articles/9298760-third-party-packages).

Is the flatpak support is official or not, and if yes, why is it hidden in third party packages ?
Why is it not the default considering it's better and more compatible than the appimage (it will update, it really installs, it has desktop integrations, etc) ?

**Comment by CompeyDev** [comment]
> @CompeyDev is working on a package

@intergrav This isn't true - I said I would be willing to work on one if required.

**Comment by intergrav** [comment]
@CompeyDev fixed :)

**Comment by getchoo** [comment]
> Is the flatpak support is official or not, and if yes, why is it hidden in third party packages ?

it is not, which is why it's placed there

**Comment by Bobrokus** [comment]
I'm suggesting my improved version of brawaru's Scoop package manifest: https://github.com/Bobrokus/grain/blob/master/bucket%2Fmodrinth-app.json

I might also ask for merging into the official `Games` Scoop bucket

**Comment by brawaru** [comment]
> `"Remove-Item $env:Appdata/com.modrinth.theseus, $env:LocalAppdata/com.modrinth.theseus -Force -Recurse -ErrorAction Ignore"`

This is highly concerning. You should not get rid of `%AppData%\com.modrinth.theseus` if you don't use it. It may actually contain user's profiles.

> `"THESEUS_CONFIG_DIR": "$dir/data"`

This will have an unwanted side of effect of forcing every Modrinth App instance running on your Windows account to use the persist directory for Scoop installation of Modrinth App.

**Comment by Bobrokus** [comment]
> > `"Remove-Item $env:Appdata/com.modrinth.theseus, $env:LocalAppdata/com.modrinth.theseus -Force -Recurse -ErrorAction Ignore"`
> 
> This is highly concerning. You should not get rid of `%AppData%\com.modrinth.theseus` if you don't use it. It may actually contain user's profiles.
> 
> > `"THESEUS_CONFIG_DIR": "$dir/data"`
> 
> This will have an unwanted side of effect of forcing every Modrinth App instance running on your Windows account to use the persist directory for Scoop installation of Modrinth App.

1. Can be overcomed by creating script that will migrate the data from %appdata% if it contains some
2. I understand your concern but I don't think it's very likely the user will have 2 installations. But if you think it's a big deal, I think there could be a warning that the var has been changed and add a guide how to change it back

**Comment by Bobrokus** [comment]
> > `"Remove-Item $env:Appdata/com.modrinth.theseus, $env:LocalAppdata/com.modrinth.theseus -Force -Recurse -ErrorAction Ignore"`
> 
> This is highly concerning. You should not get rid of `%AppData%\com.modrinth.theseus` if you don't use it. It may actually contain user's profiles.
> 
> > `"THESEUS_CONFIG_DIR": "$dir/data"`
> 
> This will have an unwanted side of effect of forcing every Modrinth App instance running on your Windows account to use the persist directory for Scoop installation of Modrinth App.

1. We can either ask the user if he would like to migrate or keep the data
2. I don't think this is a very likely scenario imo but if it's a big deal then we can warn the user about the change and show him how to change it back. But personally I don't like the `.bat` approach. Just seems to complicated.

Either way, thanks for the useful feedback, and let me know what you think!

**Comment by Sid127** [comment]
I have packaged modrinth-app in a Fedora Copr, available here: https://copr.fedorainfracloud.org/coprs/sid127/packages

**Comment by mirai-toto** [comment]
Hello, I have read the entire discussion, but I am still unsure why Flatpak is not officially supported. Is it due to a technical issue or a lack of resources?

**Comment by ArduFish123** [comment]
What about snaps?

**Comment by MagmaBro123** [comment]
> What about snaps?

Snaps are not being dealt with by anyone as far as I know.

**Comment by CompeyDev** [comment]
Ew, Snaps.

**Comment by Norbiros** [comment]
I uploaded `modrinth-app-bin` to Gentoo's GURU repository.

**Comment by intergrav** [comment]
> I uploaded `modrinth-app-bin` to Gentoo's GURU repository.

added to the issue

**Comment by CyberCatStudios** [comment]
Just gonna say that the AUR version is very buggy. Things like opening logs or clicking out of a context menu have tendencies to crash. The only other option, which is flatpak, has glitched GUI making it impossible to sign into a Microsoft account.

**Comment by MagmaBro123** [comment]
> Just gonna say that the AUR version is very buggy. Things like opening logs or clicking out of a context menu have tendencies to crash. The only other option, which is flatpak, has glitched GUI making it impossible to sign into a Microsoft account.

Most are aware about the Flatpak version's issues. Webkitgtk is... really unfriendly with Nvidia cards. Tauri, what odrinth uses, needs to provide other web options (like Chromium or Firefox's Geko) in order for this to be dealt with properly

**Comment by Enderteck** [comment]
I'm on an AMD gpu and I'm still getting webkitGTK crashes with the modrinth app, it's not an Nvidia exclusive issue.

**Comment by MagmaBro123** [comment]
Still. Tauri should really consider providing other options besides Webkit.

**Comment by Norbiros** [comment]
We are working on CEF (Chromium Embedded Framework) support [here](https://github.com/wusyong/cef-rs). But it still may take some time. I found out that compiling manually webkit and this app fixes a lot of issues and improves the performance. Also almost all packaging options are way better than AppImage...

**Comment by MagmaBro123** [comment]
Glad to see that. But as I said though, Webkitgtk is horrible with Nvidia.

**Comment by TheBossMagnus** [comment]
I've been maintaining the modrnth app manifest on WinGet until now, but there's an issue-l no longer use Windows and can't test it. For the time being, I can keep updating it from Linux using a third-party tool, but I can't guarantee its functionality since I can't test it (though it's very unlikely to break). For the time being I can keep updating it np, but this isn't ideal...

I'm looking for someone to take over maintenance. It's not complicated. It's just a CLI command to update the manifest. 

Alternatively, this could be automated using GitHub Actions to remove the need for manual updates. 


@intergrav pls update the Winget infos

**Comment by Enderteck** [comment]
Hey I can't download the .rpm modrinth app from the website, is this expected ?

This is the link from https://modrinth.com/app but it leads to an error page

https://launcher-files.modrinth.com/versions/0.10.3/linux/Modrinth%20App_0.10.3-1.x86_64.rpm

**Comment by WhiteDayZ** [comment]
👆 same issue

**Comment by Prospector** [comment]
RPM link has been fixed

**Comment by Prospector** [comment]
or at least, it will be when the cache expires.

Correct link is: https://launcher-files.modrinth.com/versions/0.10.3/linux/Modrinth%20App-0.10.3-1.x86_64.rpm

**Comment by Enderteck** [comment]
BTW there is no icon for the modrinth app in the taskbar when launched as there is no StartupWMClass in Modrinth App.desktop.
Easily fixed by adding

StartupWMClass=ModrinthApp

to Modrinth App.desktop

**Comment by Enderteck** [comment]
> Hey I can't download the .rpm modrinth app from the website, is this expected ?
> 
> This is the link from https://modrinth.com/app but it leads to an error page
> 
> https://launcher-files.modrinth.com/versions/0.10.3/linux/Modrinth%20App_0.10.3-1.x86_64.rpm

Happening again

**Comment by ArduFish** [comment]
Deb and Rpm are working, Appimage is not.

**Comment by Enderteck** [comment]
The rpm is definetly not working at least here in France

<img width="1765" height="971" alt="Image" src="https://github.com/user-attachments/assets/0de040bb-a0c3-404c-9fab-25a7bff36902" />

**Comment by piperswe** [comment]
Would it be possible to also get an `apt` repo set up so that Modrinth updates run through `apt` (like the rest of the system)?

**Comment by Scopomania** [comment]
I i make the modrinth apk version 

[base.zip](https://github.com/user-attachments/files/21546142/base.zip)

(No goggle play license tho)

**Comment by Khaivn1308** [comment]
> Would it be possible to also get an `apt` repo set up so that Modrinth updates run through `apt` (like the rest of the system)?

i have the same question. is there an apt repo? and no im not gonna use flatpak. flatpak is full of issues

**Comment by MagmaSlime123** [comment]
> > Would it be possible to also get an `apt` repo set up so that Modrinth updates run through `apt` (like the rest of the system)?
> 
> i have the same question. is there an apt repo? and no im not gonna use flatpak. flatpak is full of issues

No there isn't. Flatpak isn't perfect, but it's the best you'll get.

**Comment by Prospector** [comment]
Going to close this as the flatpak is now the officially supported linux distribution, I don't believe we plan to maintain any others, but people are welcome to create third party distributions.
